### PR TITLE
Implement file checks and use new architecture

### PR DESCRIPTION
**Pull request by zheileman**
This PR implements the remaining 2 file checks we had in the [previous big PR](https://github.com/ministryofjustice/repo-audit/pull/30) and starts making use of the new checks and code, removing/refactoring the superseded code.

Both scripts in the `bin` directory will now run against the new checks and making use of the configuration file, which has been updated to only leave 2 file checks: `LICENSE` and `Dangerfile`.

### CODE REVIEW

File: spec/checks/checks_shared_examples.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+require_relative '../spec_helper'
```
Comment (SteveMarshall) [inline spec/checks/checks_shared_examples.rb]: I'd like to see the factoring-out of shared examples as a separate commit, personally, so the new functionality isn't muddled in with moving-around of old.

File: spec/checks/file_content_check_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    context 'for a success result' do
+      let(:file_content) { File.read('spec/fixtures/example-license.txt') }
+
+      it 'returns the expected result' do
+        expect(subject).to receive(:success)
+        subject.run(repository)
+      end
+    end
+
+    context 'for a failure result' do
```
Comment (SteveMarshall) [inline spec/checks/file_content_check_spec.rb]: What happens when only one of the expected content blocks is matched? Can we test that?

Comment (zheileman) [inline spec/checks/file_content_check_spec.rb]: Yep, I will include a test for that scenario. It should fail.

Right now it is all or nothing, as in all the matchers need to be successful for the whole `check` to be successful (i.e. all the strings must be in the content of the file being checked). But if we were to anticipate the necessity of a check that is successful `as long as any of the given matchers are found` then we can extend the check to cope with it or create a new check for that.

File: spec/checks/checks_shared_examples.rb (hunk 1 above)
Comment (zheileman) [inline spec/checks/checks_shared_examples.rb]: I've sliced the original commit into two, so now 6766bdd will contain the refactor of the common tests to a shared file 👍

File: spec/checks/file_content_check_spec.rb (hunk 2 above)
Comment (zheileman) [inline spec/checks/file_content_check_spec.rb]: Done, I've added an extra test scenario https://github.com/ministryofjustice/repo-audit/pull/37/commits/bf14da10745daebec9508212c6b3b2f07ff49c5f#diff-ef5d2827017277f9871b7048b7630989R40

**Comment by digitalronin** [comment]
There's a lot of stuff we want to do that depends on having the checks architecture in place. All of @SteveMarshall 's changes have been addressed, so I'm going to approve this PR and dismiss Steve's review so that we can merge this.

**Pull request merged**